### PR TITLE
PR Version 1

### PR DESCRIPTION
**Pull request by debasishdebs**
It involved Code Changes and the following are the features as part of this PR:

- Added support for Serializing & De-serializing Relation Identifiers.
- Added support for querying based on Text Predicates like textFuzzy, textContains etc.
- Added Client to connect to JanusGraph which adds the JanusGraph serializers and deserializers so that is can be queried.
- Added docs for the functionalities provided
- Adds a installer script to build/install the library and supports Unix, Mac, Windows systems.

Fixing #19 #22  from previous PRs

### CODE REVIEW

**Comment by debasishdebs** [comment]
@FlorianHockmann : I still see some commits missing CLA signs. How do we handle that? Do you need me to squash all those commits into single commit?

Also, Please review this commit. Yes this does contain 54 files but in terms of functionality, its a minimal functionality libarry.

Let me know if you need anything else.

File: src/integrationtest/python/TextAttribute_tests.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    def setUp(self):
+        self.container = JanusGraphContainer()
+
+        self.client = JanusGraphClient().connect(host=self.container.get_host_ip(), port="8182",
+                                                 traversal_source="gods_traversal").get_connection()
+
+        self.container.start()
+        self.g = Graph().traversal().withRemote(self.client)
+
+    def tearDown(self) -> None:
```
Comment (FlorianHockmann) [inline src/integrationtest/python/TextAttribute_tests.py:32]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

**Comment by FlorianHockmann** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- src/main/python/janusgraph_python/core/attribute/Text.py  1
- src/main/python/janusgraph_python/structure/io/GraphsonReader.py  1
- src/unittest/python/serialization/RelationIdentifierDeserializer_test.py  1
- src/integrationtest/python/DocTraversals_tests.py  1
- src/unittest/python/structure/io/GraphsonReader_test.py  4
- src/main/python/janusgraph_python/structure/io/GraphsonWriter.py  1
- src/main/python/janusgraph_python/serializer/RelationIdentifierDeserializer.py  1
- src/main/python/janusgraph_python/core/datatypes/RelationIdentifier.py  2
- src/integrationtest/python/TextAttribute_tests.py  2
- src/main/python/janusgraph_python/driver/ClientBuilder.py  4
- src/unittest/python/structure/io/GraphsonWriter_test.py  4
- src/unittest/python/core/datatypes/RelationIdentifier_test.py  1
- src/unittest/python/serialization/RelationIdentifierSerializer_test.py  1
- src/main/python/janusgraph_python/serializer/RelationIdentifierSerializer.py  1
- src/integrationtest/python/JanusGraphContainer.py  3
- build.py  1
         

Clones added
============
- src/integrationtest/python/DocTraversals_tests.py  1
- src/integrationtest/python/TextAttribute_tests.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236&bid=16032057)

File: README.md [hunk 2]
```diff
@@ -1,6 +1,87 @@
...
+
+-   Though Installation of library is built as a functionality of the automated build scripts provided,
+      but if needed, user can follow the bellow instructions to install the library to environment of their choice.
+
+-   To install library using Pip (It is not yet hosted):
+    ```bash
+    # X is version number of JanusGraph Python client supported based on JanusGraph version chosen.
+    pip install janusgraph_python=X
+    ```
+
```
Comment (FlorianHockmann) [inline README.md:46]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

File: src/integrationtest/python/JanusGraphContainer.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        while True:
+            try:
+                client = JanusGraphClient()
+
+                client.connect(host="localhost", port="8182",
+                                                traversal_source="gods_traversal").get_connection()
+
+                client.close()
+                break
+            except ConnectionRefusedError:
```
Comment (FlorianHockmann) [inline src/integrationtest/python/JanusGraphContainer.py:50]: Codacy Issue found: [undefined name 'ConnectionRefusedError'](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

File: BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        ```bash
+        ./build.sh -d false
+        ```
+        
+    -   If you just want to build the library, and you have python interpreter installed at /usr/lib/python
+        ```bash
+        ./build.sh -d false -p /usr/lib/python
+        ./build.sh -d false -p python3.6
+        ```
+
```
Comment (FlorianHockmann) [inline BUILDING.md:73]: Codacy Issue found: [[list-item-spacing] Missing new line after list item](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

File: src/integrationtest/python/JanusGraphContainer.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                time.sleep(0.1)
+
+    def pull_image(self):
+        image = self.client.images.pull(self.BASE_IMAGE)
+        return image
+
+    def get_host_ip(self):
```
Comment (FlorianHockmann) [inline src/integrationtest/python/JanusGraphContainer.py:57]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

File: build.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pybuilder.core import use_plugin, init, Author
+
+authors = [Author("Debasish Kanhar", "dekanhar@in.ibm.com")]
+description = "Python client drivers for JanusGraph"
+copyright = "Copyright 2018 JanusGraph Python Authors"
```
Comment (FlorianHockmann) [inline build.py]: Codacy Issue found: [Redefining built-in 'copyright'](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

File: build.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+license = "Apache License v2.0"
```
Comment (FlorianHockmann) [inline build.py]: Codacy Issue found: [Redefining built-in 'license'](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

File: src/integrationtest/python/DocTraversals_tests.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    def setUp(self):
+        self.container = JanusGraphContainer()
+
+        self.client = JanusGraphClient().connect(host=self.container.get_host_ip(), port="8182",
+                                traversal_source="gods_traversal").get_connection()
+
+        self.container.start()
+        self.g = Graph().traversal().withRemote(self.client)
+
+    def tearDown(self) -> None:
```
Comment (FlorianHockmann) [inline src/integrationtest/python/DocTraversals_tests.py:31]: Codacy Issue found: [invalid syntax](https://app.codacy.com/manual/JanusGraph/janusgraph-python/pullRequest?prid=4823236)

File: src/integrationtest/python/TextAttribute_tests.py (hunk 1 above)
Comment (debasishdebs) [reply src/integrationtest/python/TextAttribute_tests.py:32]: Can be ignored.

File: src/integrationtest/python/JanusGraphContainer.py (hunk 3 above)
Comment (debasishdebs) [reply src/integrationtest/python/JanusGraphContainer.py:50]: Can be ignored. Don't know why this errors is showing up.

File: src/integrationtest/python/JanusGraphContainer.py (hunk 5 above)
Comment (debasishdebs) [reply src/integrationtest/python/JanusGraphContainer.py:57]: Not sure what this one exactly means.

File: build.py (hunk 6 above)
Comment (debasishdebs) [reply build.py]: I think this is expected right? We will be redifining those. Guess these can be ignored.

File: build.py (hunk 7 above)
Comment (debasishdebs) [reply build.py]: Same

File: src/integrationtest/python/DocTraversals_tests.py (hunk 8 above)
Comment (debasishdebs) [reply src/integrationtest/python/DocTraversals_tests.py:31]: Can be ignored.

**Review by debasishdebs — COMMENTED** [review]
@FlorianHockmann : As I had mentioned before, a few of the final Codacy issues we see now (Total 11), few of those can be ignored. These errors seems invalid. I've added comments against those so as to make is easily identifiable. 

If you agree, we can then go ahead adding custom rules for the same.

**Comment by FlorianHockmann** [comment]
Closing these 3 PRs as we have just merged another PR which contributed a first version for janusgraph-python.
@debasishdebs You are however of course more than welcome to still contribute to this library if you are still interested in that.

**Pull request closed without merging**